### PR TITLE
add transport type websockets support

### PR DESCRIPTION
**Pull request by ljonka**

### CODE REVIEW

**Comment by jpmens** [comment]
Thanks for your contribution. I like the idea, but I'd like to please have the defaults as they were before, i.e. default transport MQTT over TCP and not Websocket; that could surprise people who upgrade.

File: launcher.conf.example [hunk 1]
```diff
@@ -6,8 +6,7 @@ mqtt_clientid   = 'mqtt-launcher-1'
 mqtt_username   = None
 mqtt_password   = None
 mqtt_tls        = None              # default: No TLS
-# mqtt_username = 'jane'
-# mqtt_password = 'secret'
+mqtt_transport_type = 'websocket'   # default: 'websocket', alternative: 'tcp'
```
Comment (jpmens) [inline launcher.conf.example:9]: please default back to `tcp`

File: mqtt-launcher.py [hunk 2]
```diff
@@ -144,6 +146,9 @@ def on_disconnect(mosq, userdata, rc):
 
     if cf.get('mqtt_tls') is not None:
         mqttc.tls_set()
+        mqttc.tls_insecure_set(False)
```
Comment (jpmens) [inline mqtt-launcher.py:149]: insecure False is a bad idea even if it was that way earlier.

File: mqtt-launcher.py [hunk 3]
```diff
@@ -144,6 +146,9 @@ def on_disconnect(mosq, userdata, rc):
...
+
+    mqttc.ws_set_options(path="/ws")
```
Comment (jpmens) [inline mqtt-launcher.py:151]: I'm assuming this will have to be dependent of mqtt_transport_type

**Review by jpmens — CHANGES_REQUESTED** [review]

**Comment by ljonka** [comment]
Understandable, please take a look at the updated sugestions

**Comment by jpmens** [comment]
Looks very good to me, there's just one small thing: could we please rename `mqtt_ssl_verify` to `mqtt_tls_verify`? It's a pet pieve of mine: SSL no longer exists. :-) 

I think you'll see what I mean when looking at these two lines:

```
if cf.get('mqtt_ssl_verify') is not None:
             mqttc.tls_insecure_set(False)
```

Looks a bit awkward doesn't it? "ssl" vs. "tls" :-)

**Comment by jpmens** [comment]
Merged. Thank you!

**Pull request merged**